### PR TITLE
Updated aws-lambda

### PR DESCRIPTION
**Pull request by StefH**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

Fixed issue #10956

### CODE REVIEW

File: aws-lambda/aws-lambda.d.ts [hunk 1]
```diff
@@ -1,55 +1,61 @@
...
-        Message: string;
-        Timestamp: Date;
-    }
-
-    interface Kinesis {
-        data: string;
-    }
-
-    export interface Context {
-        log(message: string, object: any): void;
```
Comment (skarum) [inline aws-lambda/aws-lambda.d.ts]: If you remove log(), fail() and succeed() you break compatability with Lambda running node 0.1
Many still uses the old version due to some issues: http://stackoverflow.com/questions/37466847/dynamodb-causing-lambda-timeout

**Comment by skarum** [comment]
Apart from the above comment it's a nice change. And great find with the events in the aws-sdk

**Comment by StefH** [comment]
Can you please approve this one which is compatible with the latest node ?

And maybe you (or we) can create a new typing definition for the older node 0.1 ?

**Comment by skarum** [comment]
We shoud properly have a seperate version for the old version, but if you re-add the 4 lines to context it's compatible with both versions. I can look into splitting it into 2 versions later (within the next week).

**Comment by skarum** [comment]
Futhermore 4.3 is backward compatible, so you can call succeed, fail and log on context
 https://aws.amazon.com/blogs/compute/node-js-4-3-2-runtime-now-available-on-lambda/

**Comment by StefH** [comment]
OK, older functions are added.

**Comment by skarum** [comment]
Thanks

**Comment by skarum** [comment]
👍

**Comment by StefH** [comment]
@DefinitelyTyped please merge and make public.

**Comment by StefH** [comment]
@skarum When I recheck the older Node.js documentation:
http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#nodejs-prog-model-oldruntime-context-methods

I actually do see a **done** method, but this one is not present in the typings? So actually we need another update like:
`done(error: Error, result?: any): void;`

And also the fail should be like (because it should also accept an Error object):
```
fail(message: string): void;
fail(error: Error): void;

```
> 
> 
> context.done()
> 
> Causes the Lambda function execution to terminate.
> 
> Note
> 
> This method complements the succeed() and fail() methods by allowing the use of the "error first" callback design pattern.  It provides no additional functionality.
> 
> The general syntax is:
> 
> context.done(Error error, Object result);
> 
> Where:
> 
>     error – is an optional parameter that you can use to provide results of the failed Lambda function execution.
> 
>     result – is an optional parameter that you can use to provide the result of a successful function execution. The result provided must be JSON.stringify compatible. If an error is provided, this parameter is ignored. 
> 
> You can call this method without any parameters (done()), or pass null (done(null)).
> 
> AWS Lambda treats any non-null value for the error parameter as a handled exception.

**Comment by skarum** [comment]
You are 100% correct. And the context.log() method should be removed, as it is not in the documentation, I'm was convinced it was there earlier, but it's not there now.

Should I make the PR or would you like to do it?

**Comment by StefH** [comment]
Things to do:

0] Add this link (http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-using-old-runtime.html#nodejs-prog-model-oldruntime-context-methods) also to the text `Functions for compatibility with earlier Node.js Runtime v0.10.42`

1] Remove `log(message: string, object: any): void;`

2] Add `done(error: Error, result?: any): void;`

3] Add other fail: `fail(error: Error): void;`

All covered, then if you have time, please make a PR.

**Comment by StefH** [comment]
@skarum if you have time, please create/update PR, else I can also do it.

**Comment by skarum** [comment]
I'll create it, but it wil be in the weekend.

**Comment by StefH** [comment]
Perfect. Thanks.

**Comment by StefH** [comment]
@skarum 
If you update this typings, I think we should also update the new **types 2.0** as discussed in this link : https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/

These do still have the older definitions:
DefinitelyTyped : https://github.com/DefinitelyTyped/DefinitelyTyped/tree/types-2.0/aws-lambda
NPM : https://www.npmjs.com/package/@types/aws-lambda

**Comment by StefH** [comment]
@skarum Did you have time to fix this?

**Pull request merged**